internal/databroker/server_test.go

```diff
@@ -135,21 +137,45 @@ func TestServer_Query(t *testing.T) {
 	cfg := newServerConfig()
 	srv := newServer(cfg)
 
-	s := new(session.Session)
-	s.Id = "1"
-	any := protoutil.NewAny(s)
-	_, err := srv.Put(context.Background(), &databroker.PutRequest{
-		Records: []*databroker.Record{{
-			Type: any.TypeUrl,
-			Id:   s.Id,
-			Data: any,
-		}},
-	})
-	assert.NoError(t, err)
-	_, err = srv.Query(context.Background(), &databroker.QueryRequest{
-		Type: any.TypeUrl,
+	for i := 0; i < 10; i++ {
+		s := new(session.Session)
+		s.Id = fmt.Sprint(i)
+		any := protoutil.NewAny(s)
+		_, err := srv.Put(context.Background(), &databroker.PutRequest{
+			Records: []*databroker.Record{{
+				Type: any.TypeUrl,
+				Id:   s.Id,
+				Data: any,
+			}},
+		})
+		assert.NoError(t, err)
+	}
+	res, err := srv.Query(context.Background(), &databroker.QueryRequest{
+		Type: protoutil.GetTypeURL(new(session.Session)),
+		Filter: &structpb.Struct{
+			Fields: map[string]*structpb.Value{
+				"$or": structpb.NewListValue(&structpb.ListValue{Values: []*structpb.Value{
+					structpb.NewStructValue(&structpb.Struct{Fields: map[string]*structpb.Value{
+						"id": structpb.NewStringValue("1"),
+					}}),
+					structpb.NewStructValue(&structpb.Struct{Fields: map[string]*structpb.Value{
+						"id": structpb.NewStringValue("3"),
+					}}),
+					structpb.NewStructValue(&structpb.Struct{Fields: map[string]*structpb.Value{
+						"id": structpb.NewStringValue("5"),
+					}}),
+					structpb.NewStructValue(&structpb.Struct{Fields: map[string]*structpb.Value{
+						"id": structpb.NewStringValue("7"),
+					}}),
+				}}),
+			},
+		},
+		Limit: 10,
 	})
 	assert.NoError(t, err)
+
+	if assert.Len(t, res.Records, 4) {
+	}
```

Comment:
?

Reply:
meant to add some checks there. Nice catch.